src/utils/stompHelpers.ts

```diff
@@ -0,0 +1,121 @@
+import { Client, type Message, type StompSubscription } from '@stomp/stompjs';
+import type { IFrame } from '@stomp/stompjs';
+import SockJS from 'sockjs-client';
+import type { MessageResponse } from '@/hooks/useChatRoom';
+
+// STOMP 연결 설정 상수
+export const STOMP_CONFIG = {
+  // 재연결 시도 딜레이
+  RECONNECT_DELAY_MS: 1000 * 5,
+  // 하트비트 수신 딜레이(수신이 없을 때 연결 이상으로 판단)
+  HEARTBEAT_INCOMING_MS: 1000 * 4,
+  // 하트비트 송신 딜레이(송신이 없을 때 연결 이상으로 판단)
+  HEARTBEAT_OUTGOING_MS: 1000 * 4,
+} as const;
+
+const subscribeTopic = (chatRoomId: number) => {
+  return `/sub/${chatRoomId}/messages`;
+};
+
+const publishTopic = (chatRoomId: number) => {
+  return `/pub/${chatRoomId}/messages`;
+};
+
+export interface StompConnectionParams {
+  webSocketUrl: string;
+  token: string;
+  chatRoomId: number;
+  onMessageReceived: (message: MessageResponse) => void;
+  onError: (error: IFrame) => void;
+}
+
+export interface StompConnection {
+  client: Client;
+  subscription: StompSubscription;
+}
+
+/**
+ * STOMP 클라이언트를 생성하고 연결합니다.
+ */
+export const createStompConnection = ({
+  webSocketUrl,
+  token,
+  chatRoomId,
+  onMessageReceived,
+  onError,
+}: StompConnectionParams): Promise<StompConnection> => {
+  return new Promise((resolve, reject) => {
+    const socket = new SockJS(webSocketUrl);
+    const stompClient = new Client({
+      webSocketFactory: () => socket as WebSocket,
+      // 디버깅 로그 출력
+      // debug: (msg: string) => console.log('[STOMP]:', msg),
+      connectHeaders: {
+        Authorization: `Bearer ${token}`,
+      },
+      onConnect: () => {
+        try {
+          // 채팅 토픽 구독
+          const callback = (message: Message) => {
+            if (!message.body) return;
+            const newMessage: MessageResponse = JSON.parse(message.body);
+            onMessageReceived(newMessage);
+          };
+
+          const subscription = stompClient.subscribe(subscribeTopic(chatRoomId), callback);
+
+          resolve({ client: stompClient, subscription });
+        } catch (error) {
+          reject(error);
+        }
+      },
+      onStompError: (error) => {
+        onError(error);
+        stompClient.deactivate();
```

Comment:
3. 에러가 나면 아예 연결을 끊는거 같은데 이때 다시 재연결하는 로직을 추가하면 어떨까요?
처음에는 82번 라인에 있는 activate()가 그 역할을 하는것인가 싶었는데 명확하지 않은거같아 질문드립니다!

Reply:
```suggestion
  retry 횟수 제한;
  재시도 UI 토스트 메시지
```
stompClient.deactivate();를 제거하고 retry제한과 재시도 토스트 메시지 알람을 넣으면 재연결하는 로직이 될 것 같습니다. 토스트 메시지 적용하면서 반영하겠습니다!